sql/signal_handler.cc

```diff
@@ -58,49 +58,70 @@ extern const char *optimizer_switch_names[];
 
 static inline void output_core_info()
 {
-  /* proc is optional on some BSDs so it can't hurt to look */
-#if defined(HAVE_READLINK) && !defined(__APPLE__) && !defined(__FreeBSD__)
+#ifndef __WIN__
   char buff[PATH_MAX];
   ssize_t len;
   int fd;
+  if ((fd= open("/etc/os-release", O_RDONLY)) >= 0)
```

Comment:
I would check with  ` >=3 ` since `0,1,2` are related to the processes `/dev/tty` file descriptors.
If used, need to be changed on other places too.

Reply:
Also I would add some handling for  error when opening, like `no file found`.

Reply:
Not necessary tty or stdin/out/error.

```
$ strace -fe trace=file,write  -o /tmp/x.strace  sql/mariadbd --no-defaults --datadir=/tmp/${PWD##*/}-datadir --socket=/tmp/${PWD##*/}.sock --plugin-dir=${PWD}/mysql-test/var/plugins/ --verbose --skip-networking 0>&1 1>&-
..

mariadbd 100386  dan    0r   CHR                1,3      0t0       4 /dev/null
mariadbd 100386  dan    1uW  REG               0,35 12582912    1305 /tmp/build-mariadb-server-10.4-datadir/ibdata1
mariadbd 100386  dan    2u   CHR             136,12      0t0      15 /dev/pts/12
```

Now, don't want to be writing to stdout right.

Or

```
$ strace -fe trace=file,write  -o /tmp/x.strace  sql/mariadbd --no-defaults --datadir=/tmp/${PWD##*/}-datadir --socket=/tmp/${PWD##*/}.sock --plugin-dir=${PWD}/mysql-test/var/plugins/ --verbose --skip-networking 0>&1 1>&- 2>&-

mariadbd 100447  dan    0r   CHR                1,3      0t0       4 /dev/null
mariadbd 100447  dan    1uW  REG               0,35 12582912    1305 /tmp/build-mariadb-server-10.4-datadir/ibdata1
mariadbd 100447  dan    2u   REG               0,35     2947    1551 /tmp/#1551 (deleted)
```

Wonder what the stderr is?

Either way, opening a file returns a valid >=0 file descriptor. We don't test the relation to 3 elsewhere.

On error reporting, its not really an error if the file isn't there. We just have a little less information to show the user.